core/types/rollup_cost.go

```diff
@@ -309,13 +311,16 @@ func extractL1GasParamsPreEcotone(config *params.ChainConfig, time uint64, data
 }
 
 // extractL1GasParamsPostEcotone extracts the gas parameters necessary to compute gas from L1 attribute
-// info calldata after the Ecotone upgrade, but not for the very first Ecotone block.
-func extractL1GasParamsPostEcotone(data []byte) (gasParams, error) {
-	if len(data) != 164 {
+// info calldata after the Ecotone upgrade, other than the very first Ecotone block.
+func extractL1GasParamsPostEcotone(isHolocene bool, data []byte) (gasParams, error) {
+	if isHolocene && len(data) >= 4 && bytes.Equal(data[0:4], HoloceneL1AttributesSelector) {
+		if len(data) != 180 {
+			return gasParams{}, fmt.Errorf("expected 180 L1 info bytes, got %d", len(data))
+		}
+	} else if len(data) != 164 {
 		return gasParams{}, fmt.Errorf("expected 164 L1 info bytes, got %d", len(data))
```

Comment:
I think there's an edge case in which this error message may be misleading, namely if `len(data) < 4` then we enter this `else if` and say that `164` bytes were expected, even though if we're at Holocene, we would probably expect `180` bytes.

Reply:
yup changed